bin/gwd/request.ml

```diff
@@ -349,33 +349,36 @@ let propose_base conf =
   Output.print_sstring conf "</button></li></ul>";
   Hutil.trailer conf
 
-let try_plugin list conf base m =
+let try_plugin list conf base_name m =
   let fn =
     if List.mem "*" list
-    then (fun ( _, fn) -> fn conf base)
-    else (fun (ns, fn) -> (List.mem ns conf.forced_plugins || List.mem ns list) && fn conf base)
+    then (fun ( _, fn) -> fn conf base_name)
+    else (fun (ns, fn) -> (List.mem ns conf.forced_plugins || List.mem ns list) && fn conf base_name)
   in
   List.exists fn (Hashtbl.find_all GwdPlugin.ht m)
 
-let w_lock ~onerror fn conf base =
+let w_lock ~onerror fn conf (base_name : string option) =
   let bfile = Util.bpath (conf.bname ^ ".gwb") in
   Lock.control
-    (Mutil.lock_file bfile) false
-    ~onerror:(fun () -> onerror conf base)
-    (fun () -> fn conf base)
+    (Mutil.lock_file bfile) true
+    ~onerror:(fun () -> onerror conf base_name)
+    (fun () -> fn conf base_name)
 
-let w_base ~none fn conf base =
-  match base with
+let w_base ~none fn conf (bfile : string option) =
+  match bfile with
   | None -> none conf
-  | Some base ->
-    let conf = make_henv conf base in
-    let conf = make_senv conf base in
-    let conf =
-      match Util.default_sosa_ref conf base with
-      | Some p -> { conf with default_sosa_ref = get_iper p, Some p }
-      | None -> conf
-    in
-    fn conf base
+  | Some bfile ->
+     let base = try Some (Gwdb.open_base bfile) with _ -> None in
```

Comment:
It would be better to make `Gwdb.open_base` return a result

Reply:
Result is only available since 4.08. If we want to use it we need to drop support for older versions.